community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/commands/expressions/PointFunction.scala

```diff
@@ -33,14 +33,19 @@ case class PointFunction(data: Expression) extends NullInNullOutExpression(data)
       val map = mapCreator(state.query)
       map.getOrElse("crs", CRS.WGS84.name) match {
         case CRS.Cartesian.name =>
-          val x = map("x").asInstanceOf[Double]
-          val y = map("y").asInstanceOf[Double]
+          if (!map.contains("x") || !map.contains("y")) throw new SyntaxException("A cartesian point must contain 'x' and 'y' coordinates")
```

Comment:
We don't have a such a CypherException but don't you think it is a SyntaxError? `point` requires there to be an `x` and a `y`, isn't failure to have that a syntax exception? I added SemanticChecking for this as well and in that case we fail with SyntaxExceptions right?